bot/exts/utils/reminders.py

```diff
@@ -198,17 +338,41 @@ async def send_reminder(self, reminder: dict, expected_time: t.Optional[time.Tim
         jump_url = reminder.get("jump_url")
         embed.description += f"\n[Jump back to when you created the reminder]({jump_url})"
         partial_message = channel.get_partial_message(int(jump_url.split("/")[-1]))
+        reminder_id = reminder['id']
+        reminder_author_id = reminder['author']
+        snooze_button_view = SnoozeButtonView(self.bot, reminder_id, reminder_author_id)
         try:
-            await partial_message.reply(content=f"{additional_mentions}", embed=embed)
+            m = await partial_message.reply(
+                content=f"{additional_mentions}",
+                embed=embed,
+                view=snooze_button_view
+            )
         except discord.HTTPException as e:
             log.info(
                 f"There was an error when trying to reply to a reminder invocation message, {e}, "
                 "fall back to using jump_url"
             )
-            await channel.send(content=f"<@{reminder['author']}> {additional_mentions}", embed=embed)
+            m = await channel.send(
+                content=f"<@{reminder_author_id}> {additional_mentions}",
+                embed=embed,
+                view=snooze_button_view
+            )
+
+        # Before deleting the reminder, see if the user wants to "snooze" it
+        await snooze_button_view.wait()
+
+        if not snooze_button_view.reminder_was_snoozed:
```

Comment:
My comment above applies to this as well. We can use the return value of `wait()`.